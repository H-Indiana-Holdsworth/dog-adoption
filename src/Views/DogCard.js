import React from 'react';
import { useEffect, useState } from 'react';
import { getDogsById } from '../services/service-dogs';
import DogDetail from '../Components/DogDetail';
import Header from '../Components/Header';
import { deleteDog } from '../services/service-dogs';

export default function DogCard(props) {
  const [dogCard, setDogCard] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getDogsById(id);
      setDogCard(resp);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h3>Loading...</h3>;
  }

  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDog(dogCard);
    alert(`${dogCard.name} was successfully deleted`);
  };

  return (
    <div>
      <DogDetail dogCard={dogCard} handleDelete={handleDelete} />
      <Header />
    </div>
  );
}
