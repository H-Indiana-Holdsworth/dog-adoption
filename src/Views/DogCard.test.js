import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogCard from '../Views/DogCard';

test('DogCard should render header', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/5']}>
      <Route exact path="/dogs/:id">
        <DogCard match={{ params: { id: 5 } }} />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('Meet Lil Mark');

  expect(container).toMatchSnapshot();
});
