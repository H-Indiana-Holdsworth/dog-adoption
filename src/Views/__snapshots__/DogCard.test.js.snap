// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCard should render header 1`] = `
<div>
  <div>
    <div
      class="dog-card"
    >
      <h2>
        Meet 
        Barton
      </h2>
      <p>
        <img
          class="dog-img"
          src="https://placedog.net/500?id=6"
        />
      </p>
      Barton
       is a 
      1
       year old 
      Weimaraner
      <p>
        Adores every person he meets. Very social and lots of energy
      </p>
    </div>
    <header>
      <a
        href="/"
      >
        Home
      </a>
    </header>
  </div>
</div>
`;
